### PR TITLE
adds accessible autocomplete

### LINKED ISSUE #234: Create safari compatible autocomplete
At the moment the autocomplete for the search by repo functionality comes from the HTML5 `datalist` tag, which is not compatible with Safari. 

As discussed in this pr review: https://github.com/dwyl/tudo/pull/223#pullrequestreview-61858983 we decided to merge this in, and work on creating a more widely browser compatible autocomplete as a separate feature.

The autocomplete will be made with front end javascript to enhance browser compatibility.

### PR DESCRIPTION
**Pull request by ZooeyMiller**
Changes the current search input/autocomplete which is using datalist (non-safari compatible) to use the [GDS accessible-autocomplete](https://github.com/alphagov/accessible-autocomplete) which seems to work pretty nicely. 

Fix #234

### CODE REVIEW

File: web/static/css/app.css [hunk 1]
```diff
@@ -62,3 +62,123 @@ body {
...
+  padding: 4px;
+}
+@media (min-width: 641px) {
+  .autocomplete__hint,
+  .autocomplete__input,
+  .autocomplete__option {
+    font-size: 19px;
+    line-height: 1.31579;
+  }
+}
```
Comment (samhstn) [inline web/static/css/app.css]: Are you not already loading all of this css here: https://github.com/dwyl/tudo/pull/241/files#diff-4e4c242ad765559995e9844dd68aa05aR13

**Review by samhstn — CHANGES_REQUESTED** [review]

File: web/static/css/app.css (hunk 1 above)
Comment (finnhodgkin) [reply web/static/css/app.css]: I think @ZooeyMiller meant to delete this file - probably a missed `git add` for the delete..?

Comment (finnhodgkin) [reply web/static/css/app.css]: But yeah, I'll drop it now :+1:

**Review by samhstn — CHANGES_REQUESTED** [review]
I think it might be better to instead load in this css with another `link` tag rather than appending the css file content. Or maybe add this as in separate file with brunch

**Comment by finnhodgkin** [comment]
@shouston3 I was going with the principal of 'less-requests-is-better' but you're right, it's definitely clearer having an individual file. 

Does Phoenix not include support for css bundling? That would be the best of both worlds.
(I'm googleing it now)
Well turns out it's included by default... I see :laughing:

**Comment by finnhodgkin** [comment]
@shouston3 It's now a separate file included with the css bundle :+1:

**Review by samhstn — APPROVED** [review]

File: web/static/js/app.js [hunk 2]
```diff
@@ -11,11 +11,22 @@
...
 // Import local files
 //
 // Local files can be imported directly using relative
 // paths "./socket" or full ones "web/static/js/socket".
 
 // import socket from "./socket"
+//
+// const repos = ['learn-elixir', 'time', 'tudo'];
+
+accessibleAutocomplete({
```
Comment (Jbarget) [inline web/static/js/app.js]: could we make use of this?

https://github.com/alphagov/accessible-autocomplete#progressive-enhancement

File: web/static/js/app.js [hunk 3]
```diff
@@ -11,11 +11,22 @@
...
+import accessibleAutocomplete from 'accessible-autocomplete';
 
 // Import local files
 //
 // Local files can be imported directly using relative
 // paths "./socket" or full ones "web/static/js/socket".
 
 // import socket from "./socket"
+//
+// const repos = ['learn-elixir', 'time', 'tudo'];
```
Comment (Jbarget) [inline web/static/js/app.js]: can remove comment

File: web/templates/layout/app.html.eex [hunk 4]
```diff
@@ -20,6 +20,7 @@
         <%= render @view_module, @view_template, assigns %>
       </main>
     <script src="<%= static_path(@conn, "/js/app.js") %>"></script>
+
```
Comment (Jbarget) [inline web/templates/layout/app.html.eex]: can remove extra line

File: web/templates/page/index.html.eex [hunk 5]
```diff
@@ -78,3 +72,13 @@
   </div>
 </section>
 <%= pagination_link(@conn, @rummage) %>
+
+
+
+<script type="text/javascript">
+const repos = [
```
Comment (Jbarget) [inline web/templates/page/index.html.eex]: indentation

File: web/static/js/app.js (hunk 2 above)
Comment (finnhodgkin) [reply web/static/js/app.js]: For sure, yes. The current implementation includes a select menu so we won't even have to write new code. :+1:

Comment (ZooeyMiller) [reply web/static/js/app.js]: This change has been made now, please re-review when you get a chance :pray:

**Review by Jbarget — APPROVED** [review]

**Pull request merged**